autoload/go/util.vim

```diff
@@ -85,6 +85,21 @@ function! go#util#Shelllist(arglist, ...)
     endtry
 endfunction
 
+" Returns the byte offset for line a:l and column a:c
+function! go#util#Offset(l, c)
+    if &encoding != 'utf-8'
+        let buf = a:l == 1 ? '' : (join(getline(1, a:l-1), "\n") . "\n")
+        let buf .= a:c == 1 ? '' : getline('.')[:a:c-2]
+        return len(iconv(buf, &encoding, 'utf-8'))
+    endif
+    return line2byte(a:l) + (a:c-2)
+endfunction
```

Comment:
I think the newline ending are different for fileformat dos. For example in `complete.vim`'s offset implementation there was a line:

```
 let sep = &l:fileformat == 'dos' ? "\r\n" : "\n"	
```

that would check for that. Can you please go over the offset functions and also add these kind of edge cases into this funciton ?